src/backend/gl/src/window/egl.rs

```diff
@@ -217,16 +225,23 @@ impl hal::Instance<crate::Backend> for Instance {
         };
         let attributes = [
             egl::RENDER_BUFFER as usize,
-            egl::SINGLE_BUFFER as usize,
+            egl::BACK_BUFFER as usize,
             // Always enable sRGB
             egl::GL_COLORSPACE as usize,
             egl::GL_COLORSPACE_SRGB as usize,
             egl::ATTRIB_NONE,
         ];
+
+        let native_window_ptr = match has_handle.raw_window_handle() {
+            #[cfg(target_os = "android")]
+            Rwh::Android(_) => native_window as *mut _ as _,
```

Comment:
interestingly, there is similar code in the Wayland PR, so this would need to be merged upon rebase